scylla/src/client/session.rs

```diff
@@ -1585,6 +1566,54 @@ impl Session {
         Ok(prepared_batch)
     }
 
+    async fn last_minute_prepare_batch<'b>(
+        &self,
+        init_batch: &'b Batch,
+        values: impl BatchValues,
+    ) -> Result<Cow<'b, Batch>, PrepareError> {
```

Comment:
:question: What does the name of this method mean?

Reply:
I couldn't think of a good name for it and I wanted to get the idea out there rather than bikeshed on it, my bad. 

Basically this is the preparing of the batch that occurs at the moment a batch gets executed if any of its statements with  values weren't already prepared. This logic already existed but it was done in a latter step and I had to do move it here since I wanted the execution of `Batch` to be built upon `BoundBatch`  to make it easier to catch bugs instead of duplicating logic. `BoundBatch` does not allow for unprepared statements with values so this "last minute prepare" was moved here.